verifyFiles.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+
+# #verifyFiles.py
+#
+#Script to check whether the output files generated across 
+#subject folders in different conditions are the same or not.
+#The conditions on which the script will check the match between two files are 
+#filename, size, content, modification time, distance etc.
+#
+#
+#
+#
+# ##Maintainters
+#
+# * Big Data Lab Team, Concordia University.
+# * email : tristan.glatard@concordia.ca, laletscaria@yahoo.co.in
+#
+#
+
+import os
+import sys
+import subprocess
+import argparse,textwrap
+import hashlib
+from collections import defaultdict,OrderedDict
+
+#study_folder_details_dict_list is a list for storing ordered dictionaries 
+#containing the details regarding the files of  individual subjects.
+study_folder_details_dict_list=[]
+
+#Method list_files_and_dirs is used for listing files and directories 
+#present in the input directory recursively.And it returns a list of files 
+#ordered by modification time.
+#
+#Input parameter : Directory Path
+def list_files_and_dirs(dir_path):
+        list_files_and_dirs=[]
+        for dir_, _, files in os.walk(dir_path):
+           list_files_and_dirs.append(os.path.relpath(dir_,dir_path))
+           for file_name in files:
+              rel_dir=os.path.relpath(dir_,dir_path)
+              rel_file=os.path.join(rel_dir, file_name)
+              list_files_and_dirs.append(rel_file)
+	#Sort the files and subdirectories according to the modification time.
+	list_files_and_dirs.sort(key=lambda x: os.path.getmtime(os.path.join(dir_path,x)))
+        return list_files_and_dirs
+
+#Method retrieve_file_attributes returns an ordered  python dictionary 
+#to save the status details of
+#each file and directory present in the list_files_and_dirs list
+#
+#Input parameter : List contianing the details of the path of each file and directory.
+def retrieve_file_attributes(list_files_and_dirs,dir_path):
+        temp_dict=OrderedDict(defaultdict(list))
+        for rel_path in list_files_and_dirs:
+           dir_details=os.stat(os.path.join(dir_path,rel_path))
+	   temp_dict.setdefault(rel_path, []).append(dir_details)
+        return temp_dict
+
+#Method populate_study_folder_dict will store the details regarding each subject 
+#folder in an ordered python dictionary. 
+#
+#Key : Folder or file name , Value : dictionary with details of the key value
+def populate_study_folder_dict(file_path):
+	list_of_dictionaries_based_on_conditions=[]
+	#study_folders_list : List contains path to the folders contianing 
+	#the study folders based on each condition and os
+	study_folders_list=read_contents_from_file(file_path)
+	for folder in study_folders_list:
+           temp_study_folder_dict=OrderedDict()	   
+	   file_names_and_dir_array=list_files_and_dirs(folder)
+	   temp_study_folder_dict[folder]=retrieve_file_attributes(file_names_and_dir_array,folder)
```

Comment:
so you're creating another dictionary to retrieve the file attributes. Why don't you get them in the first pass, i.e. in list_files_and_dirs? It would be faster (one less for loop across all the files, remember that there can be many of such files, in the 10,000s) and use less memory (if dictionaries are large this could become a bottleneck).